qiskit_experiments/calibration_management/base_calibrations.py

```diff
@@ -0,0 +1,148 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base calibrations class."""
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Dict, Optional, Tuple, Union
+
+from qiskit.circuit import Parameter
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.calibration_management.parameter_value import ParameterValue
+from qiskit_experiments.calibration_management.calibration_key_types import (
+    ParameterKey,
+    ParameterValueType,
+)
+
+
+class BaseCalibrations(ABC):
+    """
+    TODO: add **kwargs to the methods?
+    """
+
+    # The name of the parameter under which the qubit frequencies are registered.
+    __drive_freq_parameter__ = "drive_freq"
```

Comment:
Ah okey I found these are actually used here. Why we cannot remove this restriction?
https://github.com/Qiskit-Extensions/qiskit-experiments/blob/a387675a3fe817cef05b968bbf3e05799a09aaae/qiskit_experiments/calibration_management/update_library.py#L174

Reply:
They are used here: https://github.com/Qiskit-Extensions/qiskit-experiments/blob/a387675a3fe817cef05b968bbf3e05799a09aaae/qiskit_experiments/library/calibration/frequency_cal.py#L61 and here https://github.com/Qiskit-Extensions/qiskit-experiments/blob/a387675a3fe817cef05b968bbf3e05799a09aaae/qiskit_experiments/calibration_management/calibrations.py#L174-L175 (for example). We need to have some definition of the frequency parameter names since they are not attached to a schedule. But I agree that this could be done more elegantly bu we need to design this carefully. I'm not sure if fixing this fits this PR.

Reply:
I recently wrote calibration class for qutrit, but the same situation happened to store anharmonicity. I don't think there is a reason to treat these parameters differently. Basis gate library and calibration experiment should know the parameter names (maybe this is a parameter tied to a logical channel), but I think Calibrations should be agnostic to such rule, otherwise I need to implement new subclass for every global parameters. 

I agree this refactoring doesn't fit with this PR, however, because this is for an introduction of a abstract class, we should carefully select attributes here.

Reply:
Ok removed here: [6f21049](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1200/commits/6f210497fda4f690ad0ff8b4dc58db22e2501f5a) thanks to #1202